objectstore-client/src/client.rs

```diff
@@ -26,6 +28,7 @@ impl fmt::Debug for ClientBuilder {
             .field("client", &self.client)
             .field("usecase", &self.usecase)
             .field("default_compression", &self.default_compression)
+            .field("propagate_traces", &self.propagate_traces)
```

Comment:
now that we have no secret anymore that needs "protecting" in the `Debug` impl, might as well replace this with a derive, as that is the only thing that is left here.